### PR TITLE
MOEN-28134 Fix Android to Dart Method Channel Communication Breaking when `FirebaseMessaging.onBackgroundMessage` is used

### PR DESCRIPTION
**Pull request by kkgowtham-engg-sdk**
### Jira Ticket
https://moengagetrial.atlassian.net/browse/MOEN-28134

### Description
Fix Android to Dart Method Channel Communication Breaking when `FirebaseMessaging.onBackgroundMessage` is used

### CODE REVIEW

File: cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt [hunk 1]
```diff
@@ -31,6 +33,7 @@ class MoEngageCardsPlugin : FlutterPlugin {
 
     private fun initPlugin(binaryMessenger: BinaryMessenger) {
```
Comment (arshiya-moengage) [inline cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt:34]: `initPlugin()` is called every time `onAttachedToEngine()` is triggered. This is not inline with the solution.

File: cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt [hunk 2]
```diff
@@ -62,4 +65,47 @@ class MoEngageCardsPlugin : FlutterPlugin {
...
+            "$tag onReattachedToActivityForConfigChanges() : ReAttached To Activity for Config changes"
+        }
+    }
+
+
+    companion object {
+        /**
+         * Static MethodChannel instance to avoid plugin reinitializing from Background Isolate
+         */
+        internal var methodChannel: MethodChannel? = null
```
Comment (arshiya-moengage) [inline cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt:109]: This field is not used anywhere

Comment (kkgowtham-engg-sdk) [reply cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt:112]: Updated

File: cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt (hunk 1 above)
Comment (kkgowtham-engg-sdk) [reply cards/moengage_cards_android/android/src/main/kotlin/com/moengage/flutter/cards/MoEngageCardsPlugin.kt:33]: Updated. Missed Logic

**Review by arshiya-moengage — APPROVED** [review]

**Review by arshiya-moengage — APPROVED** [review]

**Pull request merged**